tests/format/flow/conditional-types/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,455 @@
+
+type Foo3 =
+  ThingamabobberFactory extends AbstractThingamabobberFactory
+    ? GobbledygookProvider
+    : CompositeGobbledygookProvider
+ extends
+  DoubleGobbledygookProvider extends MockGobbledygookProvider
+    ? MockThingamabobberFactory
+    : ThingamabobberFactory
+
+  ? UniqueDalgametreService
+  : CompositeZamazingoResolver;
+
+================================================================================
+`;
+
+exports[`new-ternary-spec.js [babel-flow] format 1`] = `
+"Unexpected token (5:5)
+  3 | // TypeScript examples:
+  4 | type TypeName<T> =
+> 5 |   T extends string ? "string"
+    |     ^
+  6 |   : T extends number ? "number"
+  7 |   : T extends boolean ? "boolean"
+  8 |   : T extends undefined ? "undefined""
+`;
+
+exports[`new-ternary-spec.js format 1`] = `
+====================================options=====================================
+parsers: ["flow"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+// @flow
+
+// TypeScript examples:
+type TypeName<T> =
+  T extends string ? "string"
+  : T extends number ? "number"
+  : T extends boolean ? "boolean"
+  : T extends undefined ? "undefined"
+  : T extends Function ? "function"
+  : "object";
+
+type Unpacked<T> =
+  T extends (infer U)[] ? U
+  : T extends (...args: any[]) => infer U ?
+    SomeReallyLongThingThatBreaksTheLine<U>
+  : T extends Promise<infer U> ? U
+  : T;
+
+=====================================output=====================================
+// @flow
+
+// TypeScript examples:
+type TypeName<T> = T extends string
+  ? "string"
+  : T extends number
+  ? "number"
+  : T extends boolean
+  ? "boolean"
+  : T extends undefined
+  ? "undefined"
+  : T extends Function
+  ? "function"
+  : "object";
+
+type Unpacked<T> = T extends infer U[]
+  ? U
+  : T extends (...args: any[]) => infer U
+  ? SomeReallyLongThingThatBreaksTheLine<U>
+  : T extends Promise<infer U>
+  ? U
+  : T;
+
+================================================================================
+`;
+
+exports[`parentheses.js [babel-flow] format 1`] = `
+"Unexpected token (4:17)
+  2 |
+  3 | // #13275
+> 4 | type Foo<T> = T extends ((...a: any[]) => infer R extends string) ? R : never;
+    |                 ^
+  5 |"
+`;
+
+exports[`parentheses.js format 1`] = `
+====================================options=====================================
+parsers: ["flow"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+// @flow
+
+// #13275
+type Foo<T> = T extends ((...a: any[]) => infer R extends string) ? R : never;
+
+=====================================output=====================================
+// @flow
+
+// #13275
+type Foo<T> = T extends (...a: any[]) => infer R: string ? R : never;
```

Comment:
TODO: I need to fix this issue of reprinting the bound syntax. Flow recently added support for the TS syntax of extends, so I need to extend the support to prettier as well.